ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
At the AST/semantic level, yes, but in the binary encoding the difference would be that `br0` never had a child subtree while `br` always did.  It's tempting to just have this be a "binary encoding detail", but, as with ast.ml, I think it's really useful to have a direct correspondence between the AST and binary encoding rather than the binary encoding introducing something "interesting" (such as multiple opcodes for a single operator or an additional hidden arity immediate).  I appreciate that there's more to discuss here, but it seems like the only bad option is reverting #180 since that breaks post-order/bottom-up.  It seems like `br0`/`br` is a conservative way to start that doesn't preclude tuples but is also something we could change later (before v.1, but after more significant tuple experimentation) if we decide we want 1 n-ary `br` instead of `br0`/`br`.